src/Umbraco.Core/PropertyEditors/ValueConverters/TagsValueConverter.cs

```diff
@@ -12,69 +10,42 @@ namespace Umbraco.Core.PropertyEditors.ValueConverters
     [DefaultPropertyValueConverter]
     public class TagsValueConverter : PropertyValueConverterBase
     {
-        private readonly IDataTypeService _dataTypeService;
+        public TagsValueConverter()
+        { }
 
+        [Obsolete("This constructor isn't required anymore, because we don't need services to lookup the data type configuration.")]
         public TagsValueConverter(IDataTypeService dataTypeService)
-        {
-            _dataTypeService = dataTypeService ?? throw new ArgumentNullException(nameof(dataTypeService));
-        }
+        { }
 
         public override bool IsConverter(IPublishedPropertyType propertyType)
             => propertyType.EditorAlias.InvariantEquals(Constants.PropertyEditors.Aliases.Tags);
 
         public override Type GetPropertyValueType(IPublishedPropertyType propertyType)
-            => typeof (IEnumerable<string>);
+            => typeof(IEnumerable<string>);
 
         public override PropertyCacheLevel GetPropertyCacheLevel(IPublishedPropertyType propertyType)
             => PropertyCacheLevel.Element;
 
         public override object ConvertSourceToIntermediate(IPublishedElement owner, IPublishedPropertyType propertyType, object source, bool preview)
         {
-            if (source == null) return Array.Empty<string>();
-
-            // if Json storage type deserialize and return as string array
-            if (JsonStorageType(propertyType.DataType.Id))
+            var sourceString = source?.ToString();
+            if (string.IsNullOrEmpty(sourceString))
             {
-                var jArray = JsonConvert.DeserializeObject<JArray>(source.ToString());
-                return jArray.ToObject<string[]>() ?? Array.Empty<string>();
+                return Array.Empty<string>();
             }
 
-            // Otherwise assume CSV storage type and return as string array
-            return source.ToString().Split(new[] { "," }, StringSplitOptions.RemoveEmptyEntries);
-        }
-
-        public override object ConvertIntermediateToObject(IPublishedElement owner, IPublishedPropertyType propertyType, PropertyCacheLevel cacheLevel, object source, bool preview)
-        {
-            return (string[]) source;
-        }
-
-        /// <summary>
-        /// Discovers if the tags data type is storing its data in a Json format
-        /// </summary>
-        /// <param name="dataTypeId">
-        /// The data type id.
-        /// </param>
-        /// <returns>
-        /// The <see cref="bool"/>.
-        /// </returns>
-        private bool JsonStorageType(int dataTypeId)
-        {
-            // GetDataType(id) is cached at repository level; still, there is some
-            // deep-cloning involved (expensive) - better cache here + trigger
-            // refresh in DataTypeCacheRefresher
```

Comment:
This isn't the case anymore in V8, so lets keep it simple and remove this unnecessary caching.